src/Datasource/ConnectionRegistry.php

```diff
@@ -79,8 +79,10 @@ protected function _create(object|string $class, string $alias, array $config):
         if (is_string($class)) {
             unset($config['className']);
 
-            /** @var \Cake\Datasource\ConnectionInterface */
-            return new $class($config);
+            /** @var \Cake\Datasource\ConnectionInterface $connection */
+            $connection = new $class($config);
```

Comment:
Why adding an unnecessary variable assignment? Phpstan doesn't understand the exisr8ng `@return` annotation?

Reply:
Unfortunately not in those cases.

Reply:
The use the return annotation only as an expected value. If they can't determine the variable type used in the return, it warns.

Reply:
Changing the existing annotation to `/** @var class-string<\Cake\Datasource\ConnectionInterface> $class */` keeps both phpstan and psalm happy and avoids any additional code.

Reply:
never mind my comment, I was thinking of a different return problem.

Reply:
Feel free to push additional commits if you feel strongly about this. It seemed a bit more easier this way.

Reply:
Adding unnecessary extra code to be cleanup later doesn't make sense. Just undo this (and similar changes) if you don't want to change them. I / someone else can deal with them later.

Reply:
My problem was that once I added them, psalm complained about the types not being consistent with the parent class.
After minutes of trying around I found the way both tools didnt complain anymore.
Since the code is still the same I didnt see the need to further check options.